src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -2174,43 +2198,18 @@ private AssemblySymbol GetForwardedToAssembly(string fullName, int arity, Diagno
                 while (current != null);
             }
 
-            // NOTE: This won't work if the type isn't using CLS-style generic naming (i.e. `arity), but this code is
-            // only intended to improve diagnostic messages, so false negatives in corner cases aren't a big deal.
-            var metadataName = MetadataTypeName.FromFullName(fullName, useCLSCompliantNameArityEncoding: true, forcedArity: arity);
-
-            var containingAssembly = this.Compilation.Assembly;
+            name = MetadataHelpers.ComposeAritySuffixedMetadataName(name, arity);
```

Comment:
>name [](start = 12, length = 4)

Please don't assign over the parameter a new value that has a different meaning. I recommend `var metadataName = ...`